recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,546 @@
+        elif self._is_py3:
+            # Python 3.xx does not support following options
+            del self.options.with_bsddb
+            del self.options.unicode
+
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            if self.settings.compiler == "Visual Studio" and "MT" in str(self.settings.compiler.runtime):
+                raise ConanInvalidConfiguration("cpython does not support MT(d) runtime when building a shared cpython library")
+        if tools.is_apple_os(self.settings.os):
+            if self._is_py2:
+                # FIXME: python2 does not build on Macos due to a missing uuid_string_t type
+                raise ConanInvalidConfiguration("This recipe (currently) does not support building python2 for apple products.")
+        if self.settings.compiler == "Visual Studio":
+            # The msbuild generator only works with Visual Studio
+            self.generators.append("msbuild")
+            if self.options.optimizations:
+                raise ConanInvalidConfiguration("This recipe does not support optimized MSVC cpython builds (yet)")
+                # FIXME: should probably throw when cross building
+                # FIXME: optimizations for Visual Studio can be added by, before building the final `build_type`:
+                # 1. build the MSVC PGInstrument build_type,
+                # 2. run the instrumented binaries, (PGInstrument should have created a `python.bat` file in the PCbuild folder)
+                # 3. build the MSVC PGUpdate build_type
+            if self.settings.build_type == "Debug" and "d" not in self.settings.compiler.runtime:
+                raise ConanInvalidConfiguration("Building debug cpython requires a debug runtime (Debug cpython requires _CrtReportMode symbol, which only debug runtimes define)")
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("MSVC does not support a static build")
+            if self._is_py2:
+                if self.settings.compiler.version >= tools.Version("14"):
+                    self.output.warn("Visual Studio versions 14 and higher are not supported")
+            if str(self.settings.arch) not in self._msvc_archs:
+                raise ConanInvalidConfiguration("Visual Studio does not support this architecure")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("Python-{}".format(self.version), self._source_subfolder)
+
+    @property
+    def _with_libffi(self):
+        return self.settings.compiler != "Visual Studio" or tools.Version(self.version) >= "3.8"
+
+    def requirements(self):
+        self.requires("openssl/1.1.1h")
+        if not (self.settings.compiler == "Visual Studio" and tools.Version(self.version) >= tools.Version("3.8")):
+            self.requires("expat/2.2.9")
```

Comment:
```suggestion
            self.requires("expat/2.2.10")
```

Reply:
Also, why is expat conditional? Shouldn't the vendored copy of expat be removed for versions >= 3.8 as well?

Reply:
I did this in dcc0b8be65ea593a208b5a6ca5927c4040bcd308
I don't remember exactly, but I think it was something about cpython using a patches expat library on msvc.
Maybe I should try it again.

Reply:
I think I fixed it. Let's see what c3i thinks about it 🙈